datafusion/catalog/src/stream.rs

```diff
@@ -321,24 +324,26 @@ impl TableProvider for StreamTable {
 
     async fn scan(
         &self,
-        _state: &dyn Session,
+        state: &dyn Session,
         projection: Option<&Vec<usize>>,
         _filters: &[Expr],
         limit: Option<usize>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        let projected_schema = match projection {
-            Some(p) => {
-                let projected = self.0.source.schema().project(p)?;
-                create_ordering(&projected, &self.0.order)?
-            }
-            None => create_ordering(self.0.source.schema(), &self.0.order)?,
+        let schema = self.0.source.schema();
+        let df_schema = DFSchema::try_from(Arc::clone(schema))?;
```

Comment:
I wonder if (re)creating this DFSchema is necessary -- it feels like at this point we know the schema information 

However, i also see we need to have a DFSchema to correctly create arbitrary PhysicalExprs so this is probably fine

Reply:
I was a bit concerned about the waste here as well, but I couldn't figure out a simple way to avoid this.